### ISSUE TITLE
Feature: aborting previous ajax request in favour of new one.

### ISSUE THREAD

**Issue by pavlo-tk** [wenzhixin/bootstrap-table#1790]
Hello, wenzhixin.
I proposing a little addition to the code that allows aborting previous ajax request in favour of new one.
It is useful, for example, on search via ajax, when user has input more one letter but previous request still processing and we no longer need it.

For this we need to modify `BootstrapTable.prototype.initServer` function
(https://github.com/wenzhixin/bootstrap-table/blob/master/dist/bootstrap-table.js#L1693):

``` javascript
    var $request;

    BootstrapTable.prototype.initServer = function (silent, query) {
    ...

        if (this.options.ajax) {
            calculateObjectValue(this, this.options.ajax, [request], null);
        } else {
            if ($request) {
                $request.abort();
            }
            $request = $.ajax(request);
            //$.ajax(request); ← old code
        }
    };
```

Take a look on this and add to the code, maybe with some refactoring.

**Comment by wenzhixin** [comment]
Thanks, I think it is useful.

**Comment by dabros** [comment]
ref: #1646 - bit of a big thread, but one of the core issues was pretty much this, the ability to selectively abort requests

**Comment by wenzhixin** [comment]
Fixed in #1977
